airflow/serialization/serializers/datetime.py

```diff
@@ -62,12 +68,30 @@ def deserialize(classname: str, version: int, data: dict | str) -> datetime.date
 
     from pendulum import DateTime
     from pendulum.tz import timezone
+    from pytz.reference import USTimeZone
```

Comment:
I think this can be moved below the version check (i.e. being part of v1 only). I am wondering if we could do this without importing pytz as we do not rely on an explicit import of pytz in Airflow